astropy/units/tests/test_distribution.py

```diff
@@ -0,0 +1,296 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from ... import units as u
+from ...utils import NumpyRNGContext
+from ...tests.helper import assert_quantity_allclose, pytest
+
+try:
+    from scipy.stats import norm  # pylint: disable=W0611
+    SMAD_FACTOR = 1 / norm.ppf(0.75)
+except ImportError:
+    HAS_SCIPY = False
+else:
+    HAS_SCIPY = True
+
+
+def test_numpy_init():
+    # Test that we can initialize directly from a Numpy array if we provide a unit
+    rates = np.array([1, 5, 30, 400])[:, np.newaxis]
+    parr = np.random.poisson(rates, (4, 1000))
+    u.Distribution(parr, unit=u.kpc)
```

Comment:
Why would this fail without a unit? Also, why have an unphysical example in the test? (Poisson is for counts, i.e., unitless).